keras_nlp/src/models/mobilenet/mobilenet_backbone.py

```diff
@@ -0,0 +1,456 @@
+
+    Example:
+    ```python
+    input_data = tf.ones(shape=(8, 224, 224, 3))
+
+    # Randomly initialized backbone with a custom config
+
+    model = MobileNetBackbone(
+        stackwise_expansion = [1, 4, 6],
+        stackwise_filters = [4, 8, 16],
+        stackwise_kernel_size = [3, 3, 5],
+        stackwise_stride = [2, 2, 1],
+        stackwise_se_ratio = [ 0.25, None, 0.25],
+        stackwise_activation = ["relu", "relu", "hard_swish"],
+        include_rescaling = False,
+        version = 'v3'
+    )
+    output = model(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        stackwise_expansion,
+        stackwise_filters,
+        stackwise_kernel_size,
+        stackwise_stride,
+        stackwise_se_ratio,
+        stackwise_activation,
+        include_rescaling,
+        input_shape=(224, 224, 3),
+        alpha=1.0,
+        version="v3",
+        **kwargs,
+    ):
+        # === Functional Model ===
+        if version not in ["v1", "v2", "v3"]:
+            raise ValueError(
+                "The `version` argument should be either `v1` (for MobileNet)"
+                "or `v2` ( for MobileNetV2)"
+                "or v3 (MobileNetV3), default version is `v3`"
+                f"Received `version={version}`"
+            )
+        inputs = keras.layers.Input(shape=input_shape)
+        x = inputs
+
+        if include_rescaling:
+            x = keras.layers.Rescaling(scale=1 / 255)(x)
+
+        first_ch = (
+            32
+            if version == "v1"
+            else (
+                adjust_channels(32 * alpha) if version == "v2" else 16
+            )  # This is for 'v3'
+        )
+        x = keras.layers.Conv2D(
+            first_ch,
+            kernel_size=3,
+            strides=(2, 2),
+            padding="same",
+            use_bias=False,
+            name="Conv",
+        )(x)
+        x = keras.layers.BatchNormalization(
+            axis=CHANNEL_AXIS,
+            epsilon=BN_EPSILON,
+            momentum=BN_MOMENTUM,
+            name="Conv_BatchNorm",
+        )(x)
+
+        if version == "v3":
+            x = apply_hard_swish(x)
+        else:
+            x = keras.layers.ReLU(6.0)(x)
+
+        for stack_index in range(len(stackwise_filters)):
+
+            x = apply_inverted_res_block(
+                x,
+                expansion=stackwise_expansion[stack_index],
+                filters=adjust_channels(
+                    (stackwise_filters[stack_index]) * alpha
+                ),
+                kernel_size=stackwise_kernel_size[stack_index],
+                stride=stackwise_stride[stack_index],
+                se_ratio=(
+                    stackwise_se_ratio[stack_index] if version == "v3" else 0
+                ),
+                activation=stackwise_activation[stack_index],
+                expansion_index=0 if version == "v1" else stack_index,
+                version=version,
+            )
+
+        if version == "v3":
```

Comment:
same here, maybe something like `output_filers=None`. if `None`, no conv and batch norm applied, if set add a final conv and batch norm